integration/truncindex_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+func genTruncIndex(normalName string) string {
+	return normalName[:(len(normalName)+1)/2]
```

Comment:
It makes sense for the unit tests to do this.. but for integration testing why use a fake TruncIndex? Why not use the vendor package "github.com/docker/docker/pkg/truncindex"?

Reply:
Sorry. I maybe do not understand what you mean.
I need to test if we can get something by its truncated name. so I make genTruncIndex function for getting its truncated name by regular name. The genTruncIndex is only for getting prefix of regular name. So the genTruncIndex is different with "github.com/docker/docker/pkg/truncindex".

Reply:
I guess I'm trying to understand why the prefix being selected is normalName[:(len(normalName)+1)/2]. It feels like the truncated name is being faked. I see that pretty much anything will work when there are only a couple items in the index. You could probably use normalName[:1] and succeed this bucket :-) The lookup merely needs to be a unique prefix in the patricia tree for the working test and a non-unique prefix for a failing test.  Maybe if we forced a failing test and a successful test this bucket would make more sense?

Reply:
Is there some kubelet, docker, or patricia tree rule that truncated name is the sliced prefix normalName[:(len(normalName)+1)/2]?

Reply:
Put another way.. what are the valid and invalid  use cases that we can expect from calls to these services apis.

Reply:
The genTruncIndex is confused, I will improve it.  And I will a invalid use case.